pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1366,6 +1366,12 @@ self: super: {
     pkgs.postgresqlTestHook
   ] super.users-postgresql-simple;
 
+  # Need https://github.com/obsidiansystems/gargoyle/pull/45
```

Comment:
Originally tried to `appendPatch` this, but the .patch derived from the commit uri didn't work, presumably because it targets the entire repo and not just the package folder.
Is there an ergonomic way to "distribute" patches to multiple libraries of a monorepo?

Reply:
You can grep for the "relative" parameter to extract the patch for a subfolder.

Reply:
Hmm that seems to require manually specifying a hash for each subpatch, so when there's N patches and M libs that's N*M hashes.
I don't see that being used often or even feasible for something like the servant repo, let alone amazonka/gogol.

What I had in mind would be something more like a "relative" parameter but for `applyPatch`/`appendPatch`, e.g.

```nix
let 
  applyPatchRelative = ???;
  mainpatch = fetchPatch { ... };
  subpatch = relative: applyPatchRelative mainpatch relative;
in {
  gargoyle = subpatch "gargoyle" super.gargoyle;
  gargoyle-postgresql = subpatch "gargoyle-postgresql" super.gargoyle-postgresql;
  gargoyle-postgresql-nix = subpatch "gargoyle-postgresql-nix" super.gargoyle-postgresql-nix;
  gargoyle-postgresql-connect = subpatch "gargoyle-postgresql-connect" super.gargoyle-postgresql-connect;
}
```